### ISSUE TITLE
Use minecraft-data to unit test packet structures

### ISSUE THREAD

**Issue by ammaraskar** [ammaraskar/pyCraft#94]
The project at https://github.com/PrismarineJS/minecraft-data/ provides machine readable information about the protocol. It might be interesting to unit test the packet structures we have against this repo to validate them.

This might help fight issues like https://github.com/ammaraskar/pyCraft/commit/860628f64bad7b40d9db5078f532732babaae48d and https://github.com/ammaraskar/pyCraft/commit/9765e936c9c2558022d361d534769f6d87ccd902

**Comment by TheSnoozer** [comment]
Would be happy to help so I could add more packets to test :-P

**Comment by yunfan** [comment]
yes and i saw they'd provided a python version of that, also there's a project to use that python version for implment mcbot, except they stop developing from 2 years ago. what a pain :[

**Comment by roblabla** [comment]
The python package is unfortunately old and outdated, but the data is really just a bunch of JSON. The wrapper packages are mostly there for convenience. It just puts the data in indexed data structures (such as dict in python) instead of arrays, using either id or name as the index.

**Comment by roblabla** [comment]
For the protocol JSON, they can be compiled to a parser in various language. We have a generator for JavaScript, Python and Rust. https://github.com/ProtoDef-io/protodefc. We also have an interpretor in JS, which can be found at https://github.com/ProtoDef-io/node-protodef with various docs, which might help understand it better.
